minio/minio
# Change behavior of service account empty policies

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Changes behavior of AddServiceAccount, UpdateServiceAccount, and InfoServiceAccount so that "empty policies" (as in policies that parse to Version = "" and no policy statements) are treated as implied policies. This is consistent how the permissions treat such a policy.

## Motivation and Context
Adding a service account in console without changing the policy leads to an empty policy that functions as an implied policy but is considered embedded, this is true of any empty policy.

## How to test this PR?
Create a service account in console without restricting policy, check the policy in mc or console

Create a service account in mc with a policy with Version = "" and Statement = [], this will convert to implied policy

## Types of changes

## Checklist:
